File: LeetCode/sort_colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        vector<int> colors(3, 0);
        
        for(int i = 0; i < nums.size(); i++) {
            colors[nums[i]]++;
        }
        
        int j = 0;
        for(int i = 0; i < 3; i++) {
            while(colors[i]--)
                nums[j++] = i;
        }
    }
};

File: LeetCode/sort_colors.go
func sortColors(nums []int)  {
    colors := [3]int{0, 0, 0}

    for _, num := range nums {
        colors[num]++
    }

    index := 0
    for i := 0; i < len(colors); i++ {
        for colors[i] > 0 {
            nums[index] = i
            index++
            colors[i]--
        }
    }
}
